.github/workflows/integration.yaml

```diff
@@ -33,28 +34,45 @@ jobs:
         run: mvn -B clean verify
 
       - name: Login to Quay
-        if: github.event_name == 'push' && github.ref_name == 'main'
+        if: github.event_name == 'push'
         uses: docker/login-action@v3
         with:
           registry: "${{ secrets.IMAGE_REPO_HOSTNAME }}"
           username: "${{ secrets.IMAGE_REPO_USERNAME }}"
           password: "${{ secrets.IMAGE_REPO_PASSWORD }}"
-
+      
       - name: Build Image
-        if: github.ref_name != 'main'
+        if: github.event_name != 'push'
         uses: docker/build-push-action@v6
         with:
           context: data-generator/
           platforms: linux/amd64,linux/arm64
           push: false
           file: data-generator/Dockerfile
 
+      - name: Image metadata
+        if: github.event_name == 'push'
+        id: meta
+        uses: docker/metadata-action@v5
+        with:
+          images: |
+            ${{ secrets.IMAGE_REPO_HOSTNAME }}/${{ secrets.IMAGE_REPO_NAMESPACE }}/flink-examples-data-generator
+          tags: |
+            type=ref,event=branch
+            type=ref,event=tag
+            type=raw,value=latest,enable={{is_default_branch}}
+          flavor: |
+            latest=false
```

Comment:
by default it would only tag an image with `latest` if it was a tag being pushed. I thought it might be better to keep the existing behaviour of tagging main pushes and come back to it.

https://github.com/docker/metadata-action?tab=readme-ov-file#latest-tag